h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -1466,6 +1437,20 @@ else if(_parms._lambda_search) { // lambda search prefers coordinate descent
     final BetaConstraint bc = _state.activeBC();
     double [] beta = _state.beta().clone();
     int numStart = activeData.numStart();
+      double temp;
+      if (_parms._family.equals(Family.binomial) || _parms._family.equals(Family.multinomial)) {
+        for (int i = 0; i < P; ++i) {
+          temp = xx[i][i] + l2pen;
+          diagInv[i] = 1.0 / temp;
+          grads[i] += temp * beta[i];
+        }
+        grads[P] += (xx[P][P]) * beta[P];
+      } else { // no change for regression
```

Comment:
I am converting the gradient calculated in equation 4.24 of ESLII-print4 into last line of equation 4.26. Now, the gradient is transpose(X)Wz. This is what Tomas used in his calculation in later parts.  By this transformation, I don't need to do any extra work.

Reply:
Tiny observation: you can calculate `xx[i][i] + l2pen;` once before the `ifelse` block starts. :)